packages/react-components/src/components/InputBoxComponent.tsx

```diff
@@ -828,23 +843,25 @@ const findMentionTagForSelection = (tags: TagData[], selection: number): TagData
 type NewSelectionIndexForMentionProps = {
   tag: TagData;
   textValue: string;
+  // current selection index
```

Comment:
maybe it's cleaner to just call these variables 'previousSelectionIndex' and 'currentSelectionIndex'?